File: src/main/java/com/gitlab/daring/image/util/ImageUtils.java
package com.gitlab.daring.image.util;

import org.bytedeco.javacpp.opencv_core.Mat;
import org.bytedeco.javacpp.opencv_core.Scalar;

import java.awt.*;

import static com.gitlab.daring.image.util.GeometryUtils.getCenterRect;
import static com.gitlab.daring.image.util.OpencvConverters.toJava;
import static com.gitlab.daring.image.util.OpencvConverters.toOpencv;

public class ImageUtils {

    public static Rectangle centerRect(Mat m, double rectSize) {
        return getCenterRect(toJava(m.size()), rectSize);
    }

    public static Mat cropMat(Mat m, Rectangle r) {
        return m.apply(toOpencv(r));
    }

    public static Mat cropCenter(Mat m, double rectSize) {
        if (rectSize == 1.0) return m;
        return cropMat(m, centerRect(m, rectSize));
    }

    public static Mat smat(int v) {
        return new Mat(new byte[]{(byte) v});
    }

    public static Scalar newScalar(String color) {
        Color c = Color.decode(color);
        return new Scalar(c.getBlue(), c.getGreen(), c.getRed(), 0);
    }

    private ImageUtils() {
    }

}

File: src/main/kotlin/com/gitlab/daring/image/util/ImageUtils.kt
package com.gitlab.daring.image.util

import com.gitlab.daring.image.opencv.rect
import com.gitlab.daring.image.opencv.sizeDim
import com.gitlab.daring.image.util.GeometryUtils.getCenterRect
import org.bytedeco.javacpp.opencv_core.Mat
import org.bytedeco.javacpp.opencv_core.Scalar
import java.awt.Color
import java.awt.Rectangle

object ImageUtils {

    fun centerRect(m: Mat, rectSize: Double): Rectangle {
        return getCenterRect(m.sizeDim, rectSize)
    }

    fun cropMat(m: Mat, r: Rectangle) = m.apply(r.rect)

    fun cropCenter(m: Mat, rectSize: Double): Mat {
        if (rectSize == 1.0) return m
        return cropMat(m, centerRect(m, rectSize))
    }

    fun smat(v: Int): Mat {
        return Mat(byteArrayOf(v.toByte()), false)
    }

    fun newScalar(color: String): Scalar {
        val c = Color.decode(color)
        return Scalar(c.blue.toDouble(), c.green.toDouble(), c.red.toDouble(), 0.0)
    }

}
